components/org.wso2.carbon.identity.application.authenticator.apple/src/main/java/org/wso2/carbon/identity/application/authenticator/apple/AppleAuthenticator.java

```diff
@@ -0,0 +1,952 @@
+        configProperties.add(scope);
+
+        Property additionalQueryParams = new Property();
+        additionalQueryParams.setName(AppleAuthenticatorConstants.ADDITIONAL_QUERY_PARAMETERS);
+        additionalQueryParams.setDisplayName("Additional Query Parameters");
+        additionalQueryParams.setValue("response_mode=form_post");
+        additionalQueryParams.setDescription("Additional query parameters to be sent to Apple.");
+        additionalQueryParams.setDisplayOrder(9);
+        configProperties.add(additionalQueryParams);
+
+        Property regenerateClientSecret = new Property();
+        regenerateClientSecret.setName(AppleAuthenticatorConstants.REGENERATE_CLIENT_SECRET);
+        regenerateClientSecret.setDisplayName("Regenerate client secret");
+        regenerateClientSecret.setType("boolean");
+        regenerateClientSecret.setValue("false");
+        regenerateClientSecret.setDescription("Specifies if the client secret should be re-generated in " +
+                "the next authentication.");
+        regenerateClientSecret.setDisplayOrder(10);
+        configProperties.add(regenerateClientSecret);
+
+        Property secretExpiryTime = new Property();
+        secretExpiryTime.setName(AppleAuthenticatorConstants.CLIENT_SECRET_EXPIRY_TIME_METADATA);
+        secretExpiryTime.setDisplayName("Client Secret Expiry Time");
+        secretExpiryTime.setDescription("Metadata property that represents the expiry time of the generated " +
+                "client secret. Don't edit this property manually.");
+        secretExpiryTime.setConfidential(true);
+        secretExpiryTime.setDisplayOrder(11);
+        configProperties.add(secretExpiryTime);
+
+        return configProperties;
+    }
+
+    /**
+     * Get authorization server endpoint.
+     *
+     * @param authenticatorProperties Authenticator properties.
+     * @return Authorization service endpoint.
+     */
+    @Override
+    protected String getAuthorizationServerEndpoint(Map<String, String> authenticatorProperties) {
+
+        if (StringUtils.isBlank(this.oAuthEndpoint)) {
+            initOAuthEndpoint();
+        }
+        return this.oAuthEndpoint;
+    }
+
+    /**
+     * Get token endpoint.
+     *
+     * @param authenticatorProperties Authenticator properties.
+     * @return Token endpoint.
+     */
+    @Override
+    protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
+
+        if (StringUtils.isBlank(this.tokenEndpoint)) {
+            initTokenEndpoint();
+        }
+        return this.tokenEndpoint;
+    }
+
+    /**
+     * Get scope.
+     *
+     * @param scope                     Scope.
+     * @param authenticatorProperties   Authenticator properties.
+     * @return Scope string.
+     */
+    @Override
+    protected String getScope(String scope, Map<String, String> authenticatorProperties) {
+
+        if (StringUtils.isBlank(scope)) {
+            return StringUtils.EMPTY;
```

Comment:
If the scope is not set, can't we set the default OIDC scope? 
```suggestion
            return OIDCAuthenticatorConstants.OAUTH_OIDC_SCOPE;
```

Reply:
Apple doesn't require sending `openid` scope. Hence it doesn't mean anything.